### PR TITLE
[SMD-32]: enable authentication

### PR DESCRIPTION
**Pull request by Joel-Pearce**
Implements basic auth.

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 
 import config
 from app import create_app
 
 
 @pytest.fixture()
-def flask_test_client() -> FlaskClient:
+def mocked_auth(monkeypatch):
+    def access_token(return_app):
+        return {"accountId": "test-user", "roles": ["COMMENTER"]}
```
Comment (gidsg) [inline tests/conftest.py:10]: Don't think we're actually using roles currently so this could be empty?

File: app/templates/base.html [hunk 2]
```diff
@@ -20,6 +20,12 @@
   {{ govukHeader({
     'homepageUrl': url_for('main.upload'),
     'serviceUrl': url_for('main.upload'),
```
Comment (gidsg) [inline app/templates/base.html:22]: Can we use the index for these instead and let them redirect? (for when we add more pages in future)

File: app/templates/base.html [hunk 3]
```diff
@@ -18,8 +18,14 @@
 
 {% block header %}
   {{ govukHeader({
-    'homepageUrl': url_for('main.upload'),
-    'serviceUrl': url_for('main.upload'),
+    'homepageUrl': url_for('main.index'),
```
Comment (Connor-OS) [inline app/templates/base.html:21]: Are we adding a seperate index page or renaming the current upload.html to index.html?

Comment (gidsg) [reply app/templates/base.html:21]: The index route currently redirects to the upload page

**Review by gidsg — APPROVED** [review]
👍 🔒

**Pull request merged**